src/sdk/service/network-validation-service/network-validation-service.ts

```diff
@@ -0,0 +1,468 @@
+    return null;
+  }
+
+  /**
+   * Indicates whether a string represents a valid IPv4 address.
+   * @param {string} ipAddress: ipAddress string representation.
+   * @returns {boolean}
+   */
+  public static isValidIPv4Address(ipAddress: string, allowReserved: boolean = true): boolean {
+    const components = ipAddress.split('.');
+    if (components.length === 4) {
+      if (!allowReserved && (components.includes('0') || components.includes('255'))) {
+        return false;
+      }
+      for (const component of components) {
+        const parsedComponent = Number(component);
+        if (component === '' || isNaN(parsedComponent) || parsedComponent % 1 !== 0 ||
+          parsedComponent < 0 || parsedComponent > 255) {
+          return false;
+        }
+      }
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Indicates whether a string represents a valid IPv4 netmask.
+   * @param {string} netmask: the netmask string to check
+   * @return {boolean}
+   */
+  public static isValidNetmask(netmask: string): boolean {
+    if (!this.isValidIPv4Address(netmask)) {
+      return false;
+    }
+    const components = netmask.split('.');
+    let found = false;
+    for (let i = 0; i < 4; i++) {
+      const val = parseInt(components[i], 10);
+      if ((val === 255 && i !== 3) || (found && val === 0)) {
+        continue;
+      } else if (found && val !== 0) {
+        return false;
+      } else if (val === 255 && i === 3) {
+        return false;
+      }
+      let template = 128;
+      for (let j = 0; j < 8; j++) {
+        if (template & val === 0) {
+          found = true;
+        } else if (found) {
+          return false;
+        }
+        template = template >>> 1;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Indicates whether a string is a valid CIDR.
+   * @param {string} cidr: cidr string
+   * @return {boolean}
+   */
+  public static isValidCidr(cidr: string): boolean {
+    const cidrComponents = cidr.split('/');
+    const prefix = cidrComponents[0];
+    const suffix = Number(cidrComponents[1]);
+    if (cidrComponents.length !== 2 || !this.isValidIPv4Address(prefix) || suffix > 31 || suffix < 1) {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Checks if string is valid subnet CIDR.
+   * @param {string} cidr: cidr string.
+   * @return {boolean}
+   */
+  public static isValidSubnetCidr(cidr: string): boolean {
+    if (!this.isValidCidr(cidr)) {
+      return false;
+    }
+    const cidrComponents = cidr.split('/');
+    const gateway = cidrComponents[0];
+    const gatewayInt = this.getIPv4AsInteger(gateway);
+    const maskSize = Number(cidrComponents[1]);
+    const revMaskInt = (2 ** (32 - maskSize)) - 1;
+    return ((gatewayInt & revMaskInt) === 0);
+  }
+
+  /**
+   * Gets the netmask and gateway/ip from a CIDR formatted network string.
+   * @param {string} cidr: cidr string.
+   * @return {IPv4NetworkDescriptor} of the form { gateway: '10.10.10.0', netmask:
```

Comment:
no need to specify the form of the return argument since we're using a statically typed return value.